### PR TITLE
refactor: simplify question answering

### PR DESCRIPTION
**Pull request by sisp**
Just a minor cleanup: I've simplified the question answering logic a little by avoiding the unnecessary creation of a list of questions that shall be asked, i.e. for which no answers have been provided by the user via CLI/API.

### CODE REVIEW

File: copier/main.py [hunk 1]
```diff
@@ -406,6 +403,7 @@ def answers(self) -> AnswersMap:
                 )
             except KeyboardInterrupt as err:
                 raise CopierAnswersInterrupt(result, question, self.template) from err
+
```
Comment (yajo) [inline copier/main.py:406]: There are people that add spaces, others remove them... let's keep only meaningful changes please 😊 

```suggestion
```

File: copier/main.py [hunk 2]
```diff
@@ -415,6 +413,7 @@ def answers(self) -> AnswersMap:
                 new_answer = question.filter_answer(new_answer)
             if new_answer != previous_answer:
                 result.user[question.var_name] = new_answer
+
```
Comment (yajo) [inline copier/main.py:416]:
```suggestion
```

**Review by yajo — CHANGES_REQUESTED** [review]
Thanks! One little change please 😇

**Comment by yajo** [comment]
Oh, BTW for you to know (because I see you're doing a lot of refactoring), I'm using commitizen to keep the changelog. It automatically adds entries to it when commits start with `refactor: `.

You probably understand that a change like this doesn't deserve an entry in the changelog, as it's just affecting the inner implementation of a function. For these changes, probably using commits with `style: ` prefix is better.

Also use `tests: ` when refactoring tests, as end users (those that read the changelog) shouldn't care about that.

For now don't worry, I will squash on merge and change it myself.

File: copier/main.py (hunk 2 above)
Comment (sisp) [reply copier/main.py:415]: Makes sense. 👍

**Comment by sisp** [comment]
Ahhh, I hadn't read your comment about the `refactor:` prefix. Will remember it next time. 😄

**Pull request merged**